opensplice_cmake_module/env_hook/opensplice.sh.in

```diff
@@ -1,10 +1,43 @@
-export OSPL_HOME=@OpenSplice_HOME@
+_BUILDTIME_OSPL_HOME=@OpenSplice_HOME@
+
+# Give priority to the OSPL_HOME value used when the workspace was built, if it's valid.
+# If it's not valid, the current value of the OSPL_HOME environment variable will be used.
+# The one detected at buildtime may refer to an invalid path for users using binaries, for example.
+if [ -d "$_BUILDTIME_OSPL_HOME" ]; then
+  # Warn about unused environment variable.
+  if [ ! -z "$OSPL_HOME" -a "$OSPL_HOME" != "$_BUILDTIME_OSPL_HOME" ]; then
+    echo "[opensplice_cmake_module] Warning: OSPL_HOME environment variable is set to [[$OSPL_HOME]]," \
+      "which is different to what it was set to when the workspace was built: [[$_BUILDTIME_OSPL_HOME]]." \
```

Comment:
going to update the wording on this because it's misleading in the case of opensplice debs:
```
[opensplice_cmake_module] Warning: OSPL_HOME environment variable is set to [[asdf]], which is different to what it was set to when the workspace was built: [[/usr]]. The latter will be used. Please make sure this is the config that you want.
```

Reply:
> Please make sure this is the config that you want.

The only problem I see with this approach is that it seems to take final control away from the end user.
Is there any way to override the build path without removing the libraries at that path if you want to explicitly use a different path without rebuilding?

Reply:
Although it's not explained in this warning, users can choose to source the vendor setup file after sourcing the ROS setup file to use non-default paths.  (this is actually what we currently require from-archive users to do because we don't have the environment variable functionality): https://github.com/ros2/rmw_opensplice/issues/218